tests/driver_at86rf215/main.c

```diff
@@ -11,208 +11,93 @@
-        for (unsigned i = 0; i < ARRAY_SIZE(keys); ++i) {
-            if (strcmp(argv[1], keys[i]) == 0) {
-                tmp = i;
-                break;
+        int res = 0;
+        if (at86rf215_subghz) {
+            res = _init_driver(&at86rf215_subghz->netdev.netdev, cb);
+            if (res) {
+                return res;
             }
         }
 
-        if (tmp == 0xFF) {
-            printf("usage: %s [freq in MHz | off]\n", argv[0]);
-            printf("valid frequencies: off");
-            for (unsigned i = 1; i < ARRAY_SIZE(keys); ++i) {
-                printf(", %s MHz", keys[i]);
+        if (at86rf215_24ghz) {
+            res = _init_driver(&at86rf215_24ghz->netdev.netdev, cb);
+            if (res) {
+                return res;
             }
-            puts("");
-            return 1;
         }
-
-        freq = tmp;
-    }
-
-    if (dev == NULL) {
-        puts("No at86rf215 radio found");
-        return 1;
     }
 
-    printf("Clock output set to %s %s\n", keys[freq], freq ? "MHz" : "");
-    at86rf215_set_clock_output(dev, AT86RF215_CLKO_4mA, freq);
-
     return 0;
 }
 
-static int cmd_get_random(int argc, char **argv)
-{
-    uint8_t values;
-    uint8_t buffer[256];
-
-    if (argc > 1) {
-        values = atoi(argv[1]);
-    }
-    else {
-        values = 16;
-    }
-
-    if (values == 0) {
-        printf("usage: %s [num]\n", argv[0]);
-        return 1;
-    }
-
-    if (dev == NULL) {
-        puts("No at86rf215 radio found");
-        return 1;
-    }
-
-    at86rf215_get_random(dev, buffer, values);
-
-    od_hex_dump(buffer, values, 0);
-
-    return 0;
-}
-
-static const shell_command_t shell_commands[] = {
-    { "batmon", "Enable the battery monitor", cmd_enable_batmon },
-    { "set_trim", "at86rf215: Set the trim value of the crystal oscillator", cmd_set_trim },
-    { "set_clko", "at86rf215: Configure the Clock Output pin", cmd_set_clock_out },
-    { "get_random", "at86rf215: Get random values from the radio", cmd_get_random },
-    { NULL, NULL, NULL }
-};
 
 int main(void)
 {
-    /* enable pktdump output */
-    gnrc_netreg_entry_t dump = GNRC_NETREG_ENTRY_INIT_PID(GNRC_NETREG_DEMUX_CTX_ALL,
-                                                          gnrc_pktdump_pid);
-    gnrc_netreg_register(GNRC_NETTYPE_UNDEF, &dump);
+    puts("Test application for AT86RF215 ethernet device driver");
 
-    /* create battery monitor thread */
-    thread_create(batmon_stack, sizeof(batmon_stack), THREAD_PRIORITY_MAIN - 1,
-                  THREAD_CREATE_STACKTEST, batmon_thread, NULL, "batmon");
+    int res = netdev_ieee802154_minimal_init();
+    if (res) {
+        puts("Error initializing devices");
+        return 1;
+    }
 
     /* start the shell */
+    puts("Initialization successful - starting the shell now");
+
     char line_buf[SHELL_DEFAULT_BUFSIZE];
-    shell_run(shell_commands, line_buf, SHELL_DEFAULT_BUFSIZE);
+    shell_run(NULL, line_buf, SHELL_DEFAULT_BUFSIZE);
```

Comment:
This application should keep the extra commands that allow to use device-specific features (such as the batmon)